### DISCUSSION TITLE
Classes not loaded from custom skeleton

### DISCUSSION THREAD

**Discussion by ThaDaVos** [orchestral/testbench#334] [Q&A] [answered]
- Testbench Version: v7.1.0
- Laravel Version: v9.2.0
- PHP Version: 8.0.16
- Database Driver & Version: SQLite

### Description:
I created a custom skeleton as I need some classes like `App/Models/User` to exist to run my tests.

I configured it as follows:

Inside `TestCase.php`
```PHP
   /**
     * Get base path.
     *
     * @return string
     */
    protected function getBasePath()
    {
        return __DIR__ . '/.skeleton';
    }
```

Inside `testbench.yaml`
```yaml
laravel: './tests/.skeleton'
```

The make commands now create everything in my custom skeleton, but the added `App\Models\User` isn't loaded during testing and throwing an error:
`Target class [App\Models\User] does not exist.`

And so you know, I am not hard depending on the user, I've got a config value for that:
```PHP
<?php

return [
    'user' => \App\Models\User::class,
];
```

And it gets loaded and used through:
```PHP
/** @var \Illuminate\Database\Eloquent\Model|\Brandlead\LaravelNovaPassportLogin\Contracts\PassportUserContract $user */
$user = App::make(PassportUserContract::class);

if (!$user instanceof Model) {
    throw new \LogicException(sprintf('%s must be an Eloquent model.', get_class($user)));
}

$user = $user::firstOrCreate(
    [
        'email' => $socialiteUser->getEmail()
    ],
    [
        'name' => $socialiteUser->getName(),
        'password' => Str::random(32),
    ]
);
```

Why do I need the user model? Simply because my package hooks into the login flow for Laravel Nova and needs to be able to create a user.

### Steps To Reproduce:
See above

**Comment by crynobone** [comment, marked as answer]
The above code just point the base_path() directory, composer still handle all the autoloading parts.

**Comment by ThaDaVos** [comment]
I solved it by adding the skeletons `app` folder to the classload as follows:
```json
"autoload-dev": {
    "psr-4": {
        "[PACKAGE_NAME_SPACE}\\Tests\\": "tests/",
        "App\\": "tests/.skeleton/app/"
    },
    "classmap": [
        "database",
        "tests/.skeleton/database"
    ]
},
```

Even the migrations from the skeleton work now
